Mathlib/Algebra/Lie/InvariantForm.lean

```diff
@@ -0,0 +1,207 @@
+  simp [orthogonal, LinearMap.BilinForm.isOrtho_def, LinearMap.BilinForm.mem_orthogonal_iff]
+
+lemma orthogonal_disjoint
+    (Φ : LinearMap.BilinForm R L) (hΦ_nondeg : Φ.Nondegenerate) (hΦ_inv : Φ.lieInvariant L)
+    -- TODO: replace the following assumption by a typeclass assumption `[HasNonAbelianAtoms]`
+    (hL : ∀ I : LieIdeal R L, IsAtom I → ¬IsLieAbelian I)
+    (I : LieIdeal R L) (hI : IsAtom I) :
+    Disjoint I (orthogonal Φ hΦ_inv I) := by
+  rw [disjoint_iff, ← hI.lt_iff, lt_iff_le_and_ne]
+  suffices ¬I ≤ orthogonal Φ hΦ_inv I by simpa
+  intro contra
+  apply hI.1
+  rw [eq_bot_iff, ← lie_eq_self_of_isAtom_of_nonabelian I hI (hL I hI),
+      LieSubmodule.lieIdeal_oper_eq_span, LieSubmodule.lieSpan_le]
+  rintro _ ⟨x, y, rfl⟩
+  simp only [LieSubmodule.bot_coe, Set.mem_singleton_iff]
+  apply hΦ_nondeg
+  intro z
+  rw [hΦ_inv, neg_eq_zero]
+  have hyz : ⁅(x : L), z⁆ ∈ I := lie_mem_left _ _ _ _ _ x.2
+  exact contra hyz y y.2
+
+end ring
+
+section field
+
+variable {K L M : Type*}
+variable [Field K] [LieRing L] [LieAlgebra K L]
+variable [AddCommGroup M] [Module K M] [LieRingModule L M]
+
+variable [Module.Finite K L]
+variable (Φ : LinearMap.BilinForm K L) (hΦ_nondeg : Φ.Nondegenerate)
+variable (hΦ_inv : Φ.lieInvariant L) (hΦ_refl : Φ.IsRefl)
+-- TODO: replace the following assumption by a typeclass assumption `[HasNonAbelianAtoms]`
+variable (hL : ∀ I : LieIdeal K L, IsAtom I → ¬IsLieAbelian I)
+
+open FiniteDimensional Submodule in
+lemma orthogonal_isCompl_coe_submodule (I : LieIdeal K L) (hI : IsAtom I) :
+    IsCompl I.toSubmodule (orthogonal Φ hΦ_inv I).toSubmodule := by
+  rw [orthogonal_toSubmodule, LinearMap.BilinForm.isCompl_orthogonal_iff_disjoint hΦ_refl,
+      ← orthogonal_toSubmodule _ hΦ_inv, ← LieSubmodule.disjoint_iff_coe_toSubmodule]
+  exact orthogonal_disjoint Φ hΦ_nondeg hΦ_inv hL I hI
+
+open FiniteDimensional Submodule in
+lemma orthogonal_isCompl (I : LieIdeal K L) (hI : IsAtom I) :
+    IsCompl I (orthogonal Φ hΦ_inv I) := by
+  rw [LieSubmodule.isCompl_iff_coe_toSubmodule]
+  exact orthogonal_isCompl_coe_submodule Φ hΦ_nondeg hΦ_inv hΦ_refl hL I hI
+
+lemma restrict_nondegenerate (I : LieIdeal K L) (hI : IsAtom I) :
+    (Φ.restrict I).Nondegenerate := by
+  rw [LinearMap.BilinForm.restrict_nondegenerate_iff_isCompl_orthogonal hΦ_refl]
+  exact orthogonal_isCompl_coe_submodule Φ hΦ_nondeg hΦ_inv hΦ_refl hL I hI
+
+lemma restrict_orthogonal_nondegenerate (I : LieIdeal K L) (hI : IsAtom I) :
+    (Φ.restrict (orthogonal Φ hΦ_inv I)).Nondegenerate := by
+  rw [LinearMap.BilinForm.restrict_nondegenerate_iff_isCompl_orthogonal hΦ_refl]
+  simp only [LieIdeal.coe_to_lieSubalgebra_to_submodule, orthogonal_toSubmodule,
+    LinearMap.BilinForm.orthogonal_orthogonal hΦ_nondeg hΦ_refl]
+  exact (orthogonal_isCompl_coe_submodule Φ hΦ_nondeg hΦ_inv hΦ_refl hL I hI).symm
+
+open FiniteDimensional Submodule in
+lemma atomistic : ∀ I : LieIdeal K L, sSup {J : LieIdeal K L | IsAtom J ∧ J ≤ I} = I := by
```

Comment:
I can't help wondering if this might hold if we just have `[IsArtinian R L]` but I really don't think you should bother trying!

Reply:
I thought about it for a bit, but we need to prove `J ⊔ J' = ⊤`, and I don't see how to do that if we're note over a field.